test/python/test_reactor.py

```diff
@@ -139,6 +139,44 @@ def test_derivative(self):
         for i in range(self.net.n_vars):
             self.assertNear(dydt[i], dy[i]/dt)
 
+    def test_finite_difference_jacobian(self):
```

Comment:
:tada:

Reply:
The semi-analytical Jacobian deliberately excludes some terms though, so I'm not sure what comparison to make that wouldn't be confusing.

Reply:
I think it would be good to have flags that will produce 'exact' Jacobians that include all terms (similar to what is done in `Kinetics` - unit tests check various versions against finite difference implementations). It is not something that needs to be done right away, but I hope to see that as part of 3.0.

What is important though is to strengthen the docstring of `Reactor::jacobian` as it is not clear what parts are omitted in the approximation?

Reply:
I can rename the method `approximateJacobian` or something similar, if you like. Generating a complete/exact Jacobian is well outside the purpose of this PR.

Reply:
Agreed that it is well outside this PR. I believe updating the docstring of `reactor::jacobian` to clarify is all that is needed (the name itself is appropriate)